### PR TITLE
Update Dockerfiles to use Alpine 3.17

### PR DESCRIPTION
**Pull request by alexku7**
**What this PR does**:
Update Dockerfiles to use Alpine 3.17

**Checklist**
- [x] Tests updated
- [x] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## master / unreleased
+* [ENHANCEMENT] Upgraded Docker base images to `alpine:3.17`. #5130
```
Comment (friedrichg) [inline CHANGELOG.md:4]: It should refer the PR # 5132 and put it further below ENHANCEMENT

**Review by friedrichg — CHANGES_REQUESTED** [review]
please also sign the DCO

**Comment by alexku7** [comment]
Done ( hopefully )

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## master / unreleased
+* [ENHANCEMENT] Upgraded Docker base images to `alpine:3.17`. #5132
```
Comment (friedrichg) [inline CHANGELOG.md:4]: nit: The order is CHANGE, FEATURE, ENHANCEMENT, BUGFIX. 
We have not been paying attention.... 😄 

can you at least put it next to the other ENHANCEMENTS?

**Review by friedrichg — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (alexku7) [reply CHANGELOG.md]: Got you :)

I moved it to the right place. Thanks

**Review by alanprot — APPROVED** [review]

**Pull request merged**